File: src/main/java/com/dv/telegram/command/ClearFailedRequests.java
package com.dv.telegram.command;

import com.dv.telegram.BotStatistics;
import com.dv.telegram.config.BotSettings;

public class ClearFailedRequests extends BasicBotCommand {

    @Override
    public String getName() {
        return ClearFailedRequests.class.getSimpleName();
    }

    @Override
    public String getDescription() {
        return String.format("%s — очистить список разных неуспешных вызовов бота с момента текущего запуска инстанса или с момента очистки этого списка.", getCommandText());
    }

    @Override
    public String getDefaultCommandName() {
        return "/clearFailedRequests";
    }

    @Override
    public String getResponse(String text, BotSettings settings, BotStatistics statistics) {
        int clearedFailedRequestsCount = statistics.failedRequests.size();
        statistics.clearFailedRequests();
        return String.format("Список из %d неудачных запросов к боту очищен.", clearedFailedRequestsCount);
    }
}

File: src/main/java/com/dv/telegram/command/ClearFailedRequests.kt
package com.dv.telegram.command

import com.dv.telegram.WikiBot

class ClearFailedRequests : BasicBotCommand() {
    override val name: String = javaClass.simpleName

    override fun getDescription(bot: WikiBot) =
        "`${bot.botName} $commandText` — очистить список разных неуспешных вызовов бота с момента текущего запуска инстанса или с момента очистки этого списка."

    override val defaultCommandName = "/clearFailedRequests"

    override fun getResponse(text: String, bot: WikiBot): String {
        val clearedFailedRequestsCount = bot.statistics.failedRequests.size

        bot.statistics.clearFailedRequests()

        return "Список из $clearedFailedRequestsCount неудачных запросов к боту очищен."
    }
}
